src/logits_processor.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+#include <future>
+#include "models/model.h"
+
+#if USE_GUIDANCE
+#include <llguidance.h>
+#endif
+
+namespace Generators {
+
+struct LogitsProcessor {
+  LogitsProcessor() = default;
+  virtual ~LogitsProcessor() = default;
+  // CommitTokens is used to commit the generated tokens to the logits processor
+  virtual void CommitTokens(std::span<int32_t> tokens) = 0;
+  // ProcessLogits is used to add logits mask to the logits
+  virtual void ProcessLogits(DeviceSpan<float> logits) = 0;
+  // Reset is used to reset the logits processor after rewinding
+  virtual void Reset() = 0;
+};
```

Comment:
Cold you add more documentation here on what the logits processor does?

Reply:
The comments are still a little vague. Could you maybe add more details on some of the internals of the logits processor. For example, given the output logits from the model, how does logits processor process them? What is expected input/output of the logits processor, and so on.